### PR TITLE
Add bzip2 port

### PR DESCRIPTION
**Pull request by msabramo**
Adds a port for bzip2.

```
root@97f995b48522:/src/cpython# emcc -s USE_BZIP2=1 bzip2.c && ls -l a.out.wasm
-rw-r--r-- 1 root root 279350 Mar 27 00:40 a.out.wasm
```

Cc: @mdboom

### CODE REVIEW

**Comment by msabramo** [comment]
I'll add a test to this soon...

File: tools/ports/bzip2.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      for src in srcs:
+        o = os.path.join(ports.get_build_dir(), 'bzip2', src + '.o')
+        shared.safe_ensure_dirs(os.path.dirname(o))
+        commands.append([shared.PYTHON, shared.EMCC, os.path.join(dest_path, src), '-O2', '-o', o, '-I' + dest_path, '-w', ])
+        o_s.append(o)
+
+      ports.run_commands(commands)
+      final = os.path.join(ports.get_build_dir(), 'bzip2', 'libbz2.a')
+      shared.try_delete(final)
+      Popen([shared.LLVM_AR, 'rc', final] + o_s).communicate()
```
[the remark this answers is not in the document]
Comment (msabramo) [reply tools/ports/bzip2.py]: Yep. Unfortunately, I made the original version of this by modifying an installed copy of the zlib port and that was an older version than what's in git now.

I updated it to be more like the zlib port in source control now.

**Comment by msabramo** [comment]
Test added.

**Comment by msabramo** [comment]
Probably want #8353 instead of this.

**Comment by kripken** [comment]
I just saw this - I think we do want this PR, and to separately land the tar.gz one? Unless there is a reason to land them together? (usually best not to, unless the features depend on each other)

**Comment by msabramo** [comment]
Yeah, I'm happy to split into separate PRs. I wouldn't merge this one as-is though, because it's using a GitHub fork repo for bzip2 since they only offered a `.tar.gz` for download on their site. This was the reason for me doing the `.tar.gz` work.

Maybe merge #8355 (I just fixed the indentation on that one) and then when that's merged, I can rebase #8353, so that it becomes a PR that just adds the bzip2 port?

**Comment by kripken** [comment]
Sounds good, yeah, can rebase that one on current incoming now.

**Comment by msabramo** [comment]
Closing in favor of #8353

**Pull request closed without merging**